x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -555,6 +555,123 @@ public void testStatsWithFilteringDefaultAliasing() {
         assertThat(Expressions.names(agg.aggregates()), contains("sum(salary)", "sum(salary) WheRe last_name ==   \"Doe\""));
     }
 
+    public void testExtractStatsCommonFilter() {
+        var plan = plan("""
+            from test
+            | stats m = min(salary) where emp_no > 1,
+                    max(salary) where emp_no > 1
```

Comment:
check the filter is recognized also when reversed e. g. where 1 < emp_no , where 1 + 2 - 5 + 4 < emp_no.

Similar to Fang's suggestion add different ORs in to check the common disjunction is extracted - I believe we already have functionality or should be easy to add it in:

`Predicates.commonFilter(List<Expression> filters)` returns Expression (common filter) plus List<Expression> - the filters list updated.

Reply:
> where 1 < emp_no , where 1 + 2 - 5 + 4 < emp_no

more tests like these added.